### ISSUE TITLE
Various compiling error after upgrading to gmp-6.1.2

### ISSUE THREAD

**Issue by zijuexiansheng** [Linuxbrew/homebrew-core#4261]
# Please fill out the issue checklist below, and provide all of the requested information.

# Please always follow these steps:
- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Linuxbrew/brew/issues/new.
- [x] If you have a macOS system and can replicate the problem on macOS with Homebrew, please report the issue to [Homebrew/core](https://github.com/Homebrew/homebrew-core) and follow their issue template from your macOS system. If you are unsure, please report the issue to Linuxbrew.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

---

My OS environment

* kernel: 3.10.0-514.26.2.el7.x86_64
* gcc: 4.8.5
* glibc: 2.17

After a clean install of linuxbrew, a minimum example of the error is as follows:

```
$ brew install glibc
$ brew install gcc
$ brew install pkg-config

==> Downloading https://pkgconfig.freedesktop.org/releases/pkg-config-0.29.2.tar.gz
######################################################################## 100.0%
==> ./configure --prefix=/home/me/linuxbrew/Cellar/pkg-config/0.29.2_1 --disable-host-tool --with-internal-glib --with
==> make
Last 15 lines from /home/me/.cache/Homebrew/Logs/pkg-config/02.make:
gvariant.c:3654:9: warning: 'my_type' may be used uninitialized in this function [-Wmaybe-uninitialized]
   value = g_variant_new_from_children (my_type,
         ^
make[6]: Leaving directory `/tmp/pkg-config-20170915-1076-1c1uaax/pkg-config-0.29.2/glib/glib'
make[5]: *** [all-recursive] Error 1
make[5]: Leaving directory `/tmp/pkg-config-20170915-1076-1c1uaax/pkg-config-0.29.2/glib/glib'
make[4]: *** [all] Error 2
make[4]: Leaving directory `/tmp/pkg-config-20170915-1076-1c1uaax/pkg-config-0.29.2/glib/glib'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/tmp/pkg-config-20170915-1076-1c1uaax/pkg-config-0.29.2/glib'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/tmp/pkg-config-20170915-1076-1c1uaax/pkg-config-0.29.2/glib'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/tmp/pkg-config-20170915-1076-1c1uaax/pkg-config-0.29.2'
make: *** [all] Error 2

READ THIS: https://github.com/Linuxbrew/brew/blob/master/docs/Troubleshooting.md#troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core, which support macOS only.
```

---

The problem was resolved by reinstalling the `gmp-6.1.1`, as follows (for anyone who wants to reinstall gmp-6.1.1 before the homebrew-core updating the gmp)

For anyone who wants to reinstall gmp-6.1.1 before the homebrew-core updates, please do the following:

* `brew edit gmp`
* change `6.1.2` to `6.1.2`
* in `bottle do`, change the sha256 value for ":x86_64_linux" to `a9b92c1ac7cc79df39c5bf7f3c2b0abeb26b9bc417c13d55e3628058aaaa301c`, and save the file
* `brew install gmp`, or `brew reinstall gmp` if you've already installed it

**Comment by jbmaggard** [comment]
Reverting to gmp-6.1.1 following instructions above seems to have fixed gfortran from the bottled gcc 5.3.0 (formula gcc).  The same bottled gfortran had started giving internal compiler errors recently, some time since my post in #4077 about having CAF (coarray fortran) working correctly with GCC 5.3.0 / MPICH 3.2 / OpenCoarrays-1.9.1.

**Comment by iMichka** [comment]
We updated the `gmp` bottle, which had a problem.
See https://github.com/Linuxbrew/brew/issues/488

Please try:
> Try brew update && brew reinstall gmp
If you get a SHA-256 error, delete the old cached bottle and try again.
